src/components/Camera/index.js

```diff
@@ -140,11 +142,11 @@ export default class Camera extends React.Component<CameraType, CameraStateType>
     this.setState({ hasGrantedPermission: true })
   }
 
-  handleWebcamFailure = () => {
-    if (this.state.hasGrantedPermission) {
-      this.setState({ hasError: true })
-    } else {
+  handleWebcamFailure = (error: Error) => {
+    if (permissionErrors.includes(error.name)) {
```

Comment:
Minor: I wonder if the react-webcam could already parse the error object and pass two parameters `originalError: Error` and `errorType: PermissionRefused | Exception`

Reply:
or at least we could have the array `permissionErrors` exported from the react-webcam module

Reply:
I thought about exposing `permissionErrors` too.

Could a third option be to add another callback prop?

- `onUserMedia`
- `onFailure`
- `onPermissionDenied`